engine/api/events.go

```diff
@@ -120,17 +84,51 @@ func (b *eventsBroker) cacheSubscribe(c context.Context, cacheMsgChan chan<- sdk
 }
 
 // Start the broker
-func (b *eventsBroker) Start(c context.Context) {
+func (b *eventsBroker) Start(ctx context.Context) {
+	b.chanAddClient = make(chan (eventsBrokerSubscribe))
+	b.chanRemoveClient = make(chan (string))
+
 	for {
 		select {
-		case <-c.Done():
-			b.cleanAll()
-			if c.Err() != nil {
-				log.Error("eventsBroker.Start> Exiting: %v", c.Err())
+		case <-ctx.Done():
+			if b.clients != nil {
+				go observability.Record(b.router.Background, b.router.Stats.SSEClients, -1*int64(len(b.clients)))
+				for c, v := range b.clients {
+					close(v.Queue)
+					delete(b.clients, c)
+				}
+			}
+			if ctx.Err() != nil {
+				log.Error("eventsBroker.Start> Exiting: %v", ctx.Err())
 				return
 			}
 		case receivedEvent := <-b.messages:
-			b.manageEvent(receivedEvent)
+			clients := make([]eventsBrokerSubscribe, len(b.clients))
+			var i int
+			for _, client := range b.clients {
+				clients[i] = client
+				i++
+			}
+			go func() {
+				for _, c := range clients {
+					log.Debug("send data to %s", c.UUID)
+					if c.Queue != nil {
+						c.Queue <- receivedEvent
```

Comment:
Maybe there can be a problem here, pushing into queue in a goroutine while this channel can also be closed by a event on chanRemoveClient.
Why not adding a mutex in the client to protect the queue, then execute all the queue actions inside routines. In fact this will separate the client from the list of clients.